Common/Data/UniverseSelection/OptionChainUniverse.cs

```diff
@@ -99,15 +99,19 @@ public override IEnumerable<Symbol> SelectSymbols(DateTime utcTime, BaseDataColl
                 return Unchanged;
             }
 
-            var availableContracts = data.Data.Select(x => x.Symbol);
+            // Temporary: this can be removed when future options universe selection is also file-based
+            var availableContractsData = data.Symbol.SecurityType != SecurityType.FutureOption
+                ? data.Data.Cast<OptionUniverse>()
+                : data.Data.Select(x => new OptionUniverse() { Symbol = x.Symbol, Underlying = data.Underlying, Time = x.Time });
+
             // we will only update unique strikes when there is an exchange date change
-            _optionFilterUniverse.Refresh(availableContracts, data.Underlying, localEndTime);
+            _optionFilterUniverse.Refresh(availableContractsData, data.Underlying, localEndTime);
 
-            var results = Option.ContractFilter.Filter(_optionFilterUniverse);
+            var results = Option.ContractFilter.Filter(_optionFilterUniverse).Cast<OptionUniverse>();
```

Comment:
`.Cast<OptionUniverse>();` should the return type of filter be of T so we can avoid the cast?
Should the `OptionFilterUniverse` be able to convert to implicit symbol enumerable? I see there's `public static implicit operator List<Symbol>(OptionFilterUniverse universe)` why list?